src/apps/Permissions/Home/AllPermissions.js

```diff
@@ -0,0 +1,150 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  DropDown,
+  GU,
+  IconSearch,
+  TextInput,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import EmptyBlock from '../EmptyBlock'
+import PermissionsView from '../PermissionsView'
+
+const ENTITY_TYPES = ['All entities', 'Accounts', 'Apps']
+
+function AllPermissions({ loading, permissions, onOpenEntity }) {
+  const [selectedEntityType, setSelectedEntityType] = useState(0)
+  const [searchTerms, setSearchTerms] = useState('')
+
+  const filteredPermissions = useMemo(() => {
+    return permissions.filter(permission => {
+      const preparedSearchTerms = searchTerms.trim().toLowerCase()
+      if (preparedSearchTerms === '') {
+        return true
+      }
+      const roleName = permission.role.name.toLowerCase()
+      const roleId = permission.role.id.toLowerCase()
+      return (
+        roleName.includes(preparedSearchTerms) ||
+        roleId.includes(preparedSearchTerms)
+      )
+    })
+  }, [searchTerms, permissions])
+
+  if (loading) {
+    return <EmptyBlock>Loading permissions…</EmptyBlock>
```

Comment:
Should there be one for empty filtered permissions too?

Reply:
Fixed in 2491f84.